catalog/app/utils/tracking.js

```diff
@@ -21,7 +23,22 @@ const consoleTracker = Promise.resolve({
   track: (evt, opts) => console.log(`track: ${evt}`, opts),
 })
 
-const mkLocation = (l) => `${l.pathname}${l.search}${l.hash}`
+function useMkLocation() {
+  const {
+    paths: { passChange: passChangePath },
+    urls: { passChange: passChangeUrl },
+  } = NamedRoutes.use()
+  return React.useCallback(
+    (l) => {
+      const pathname = matchPath(l.pathname, { path: passChangePath, exact: true })
```

Comment:
I'm curious why don't use `paths.passChange` and `urls.passChange`?

Reply:
to avoid passing whole `urls` and `paths` to memo arguments, to avoid thinking whether or not they are memoized (you can't pass just `urls.passReset`, you must pass whole `urls`, bc react eslint plugin assumes the method call uses `this`)